### PR TITLE
create target files in ./build and update .gitignore

### PR DESCRIPTION
**Pull request by kappapiana**
Creates a separate directory for outputting files and updates .gitignore to ignore that directory, so if you want to add them you need to force it .

### CODE REVIEW

File: legal/scripts/howdyadoc-legal-convert [hunk 1]
```diff
@@ -62,9 +86,10 @@ else
 fi
 
 
-echo
-echo "Executing:"
-echo $CMD
-echo
+#
```
Comment (kappapiana) [inline legal/scripts/howdyadoc-legal-convert:89]: Do we need this output?

**Comment by alpianon** [comment]
> Creates a separate directory for outputting files and updates .gitignore to ignore that directory, so if you want to add them you need to force it .

Unfortunately this is not useful - rather, it *cannot* be used if you use make to build your stuff.

In a Makefile, you have to explicitly declare the absolute or relative path of each target file (as well as of each source file), because make does need it in order to compare file dates and decide whether to recompile a certain source file or not (if source file is newer than target file, recompile it; otherwise do nothing). 
The basic Makefile syntax is

```
target(s)_path: main_source_file(s)_path secondary_source_file(s)_path
	command_to_compile_source_file(s)_into_target_file(s)
```

E.g.

```
build/%.docx: src/%.md lib/models/reference.docx common/data.yaml
	howdyadoc-legal-convert $< $@ lib/models/reference.docx
```

The above rule means:

1. targets are all `.docx` files within `build/` (one has to explicitly declare their path here!)
2. for each target file, the corresponding (main) source file is the `.md` file in `src` dir with the same file name (without extension) of the target file
3. for all target files, `lib/models/reference.docx` and `common/data.yaml` are common (secondary) source files
4. for each target file, run the command `howdyadoc-legal-convert $< $@ lib/models/reference.docx` where `$<` stands for "each main source file path" (as declared in the line above) and `$@` stands for each target source file path (as declared in the line above)

So the proposed patch would make it quite complicate to write the Makefile rule above

**Comment by kappapiana** [comment]
> > Creates a separate directory for outputting files and updates .gitignore to ignore that directory, so if you want to add them you need to force it .
> 

Brain energy to the lowest. Just cherrypick the patch to Pandoc to fix tables and forget the rest.

**Review by alpianon — COMMENTED** [review]
> Creates a separate directory for outputting files and updates .gitignore to ignore that directory, so if you want to add them you need to force it .

Unfortunately this is not useful - rather, it *cannot* be used if you use make to build your stuff.

In a Makefile, you have to explicitly declare the absolute or relative path of each target file (as well as of each source file), because make does need it in order to compare file dates and decide whether to recompile a certain source file or not (if source file is newer than target file, recompile it; otherwise do nothing). 
The basic Makefile syntax is

```
target(s)_path: main_source_file(s)_path secondary_source_file(s)_path
	command_to_compile_source_file(s)_into_target_file(s)
```

E.g.

```
build/%.docx: src/%.md lib/models/reference.docx common/data.yaml
	howdyadoc-legal-convert $< $@ lib/models/reference.docx
```

The above rule means:

1. targets are all `.docx` files within `build/` (one has to explicitly declare their path here!)
2. for each target file, the corresponding (main) source file is the `.md` file in `src` dir with the same file name (without extension) of the target file
3. for all target files, `lib/models/reference.docx` and `common/data.yaml` are common (secondary) source files
4. for each target file, run the command `howdyadoc-legal-convert $< $@ lib/models/reference.docx` where `$<` stands for "each main source file path" (as declared in the line above) and `$@` stands for each target source file path (as declared in the line above)

So the proposed patch would make it quite complicate to write the Makefile rule above

**Pull request closed without merging**